batch.md

```diff
@@ -7,37 +7,51 @@ You may obtain a copy of the License at
 
     http://www.apache.org/licenses/LICENSE-2.0
 -->
-
+# Batch Connector
 The Flink connector library for Pravega makes it possible to use a Pravega stream as a data source and data sink in a batch program.  See the below sections for details.
 
 ## Table of Contents
-- [Data Source (InputFormat)](#data-source-inputformat)
-  - [Parameters](#parameters)
+- [FlinkPravegaInputFormat](#flinkpravegainputformat)
+  - [Pravega Batch Client](#pravega-batch-client)
   - [Input Stream(s)](#input-streams)
-  - [Stream Cuts](#stream-cuts)
+  - [StreamCuts](#streamcuts)
+  - [Deserializer](#serialization)
   - [Parallelism](#parallelism)
-- [Data Sink (OutputFormat)](#data-sink-outputformat)
+  - [Builder API usage](#builder-api-usage)
+
+- [FlinkPravegaOutputFormat](#flinkpravegaoutputformat)
   - [Parameters](#parameters)
   - [Output Stream](#output-stream)
   - [Parallelism](#parallelism)
   - [Event Routing](#event-routing)
-- [Serialization](#serialization)
+  - [Serialization](#serialization)
 
-## Data Source (InputFormat)
-A Pravega stream may be used as a data source within a Flink batch program using an instance of `io.pravega.connectors.flink.FlinkPravegaInputFormat`.  The input format reads a given Pravega stream as a [`DataSet`](https://ci.apache.org/projects/flink/flink-docs-master/api/java/org/apache/flink/api/java/DataSet.html) (the basic abstraction of the Flink Batch API).   Note that the stream elements are considered to be _unordered_ in the batch programming model.
+## FlinkPravegaInputFormat
+A Pravega stream may be used as a data source within a Flink batch program using an instance of 
+```
+io.pravega.connectors.flink.FlinkPravegaInputFormat
+```
+The input format, reads a given Pravega stream as a [`DataSet`](https://ci.apache.org/projects/flink/flink-docs-master/api/java/org/apache/flink/api/java/DataSet.html) (the basic abstraction of the Flink Batch API). Note that the stream elements are considered to be **unordered** in the batch programming model.
 
-Use the [`ExecutionEnvironment::createInput`](https://ci.apache.org/projects/flink/flink-docs-master/api/java/org/apache/flink/api/java/ExecutionEnvironment.html#createInput-org.apache.flink.api.common.io.InputFormat-) method
-to open a Pravega stream as a DataSet.
+Use the following method to open a Pravega stream as a DataSet:
 
+```
+[ExecutionEnvironment::createInput](https://ci.apache.org/projects/flink/flink-docs-master/api/java/org/apache/flink/api/java/ExecutionEnvironment.html#createInput-org.apache.flink.api.common.io.InputFormat-)
 Here's an example:
 ```
-// Define the Pravega configuration
+### Pravega batch Client

```

Comment:
I actually preferred the original format where the sample code is all in one snippet. We have a separate section that talks about the APIs used in this sample code and it helps maintain the context.

Reply:
Accepted suggestion